spring-cloud-vault-config-aws/src/main/java/org/springframework/cloud/vault/config/aws/VaultConfigAwsStsBootstrapConfiguration.java

```diff
@@ -0,0 +1,163 @@
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+import org.springframework.vault.core.lease.domain.RequestedSecret;
+import org.springframework.vault.core.lease.domain.RequestedSecret.Mode;
+import org.springframework.vault.core.util.PropertyTransformer;
+
+/**
+ * Bootstrap configuration providing support for the AWS (STS) secret backend.
+ *
+ * @author Kris Iyer
+ *
+ */
+@Configuration(proxyBeanMethods = false)
+@EnableConfigurationProperties(VaultAwsStsProperties.class)
+public class VaultConfigAwsStsBootstrapConfiguration {
+
+	@Bean
+	@ConditionalOnProperty(name = "spring.cloud.vault.aws.credential-type", havingValue = "assumed_role")
+	public AwsStsSecretBackendMetadataFactory awsStsAssumedRoleSecretBackendMetadataFactory(
+			ApplicationContext context) {
+		return new AwsStsSecretBackendMetadataFactory();
+	}
+
+	@Bean
+	@ConditionalOnProperty(name = "spring.cloud.vault.aws.credential-type", havingValue = "federation_token")
+	public AwsStsSecretBackendMetadataFactory awsStsFederationTokenSecretBackendMetadataFactory(
+			ApplicationContext context) {
+		return new AwsStsSecretBackendMetadataFactory();
+	}
+
+	/**
+	 * {@link LeasingSecretBackendMetadata} for AWS integration using with STS Role
+	 * {@link VaultAwsStsProperties}.
+	 */
+	public static class AwsStsSecretBackendMetadataFactory
+			implements SecretBackendMetadataFactory<VaultAwsStsProperties> {
+
+		/**
+		 * Creates {@link LeasingSecretBackendMetadata} for a secret backend for AWS STS
+		 * using {@link VaultAwsProperties}. This accessor transforms Vault's
+		 * username/password property names to names provided with
+		 * {@link VaultAwsStsProperties#getAccessKeyProperty()}
+		 * {@link VaultAwsStsProperties#getSecretKeyProperty()} and
+		 * {@link VaultAwsStsProperties#getSessionTokenKeyProperty()}.
+		 * @param properties must not be {@literal null}.
+		 * @return the {@link LeasingSecretBackendMetadata}
+		 */
+		static LeasingSecretBackendMetadata forAws(final VaultAwsStsProperties properties) {
+
+			Assert.notNull(properties, "VaultAwsStsProperties must not be null");
+			Assert.isTrue(
+					(properties.getCredentialType().name().equals(AwsCredentialType.assumed_role.name())
+							|| properties.getCredentialType().name().equals(AwsCredentialType.federation_token.name())),
+					"Expect credential-type to be '" + AwsCredentialType.assumed_role.name() + "' or '"
+							+ AwsCredentialType.federation_token.name() + "'");
+
+			PropertyNameTransformer transformer = new PropertyNameTransformer();
+			transformer.addKeyTransformation("access_key", properties.getAccessKeyProperty());
+			transformer.addKeyTransformation("secret_key", properties.getSecretKeyProperty());
+			transformer.addKeyTransformation("security_token", properties.getSessionTokenKeyProperty());
+
+			return new LeasingSecretBackendMetadata() {
```

Comment:
Instead of adding another full config class, let's move this method (or parts of it) into `VaultConfigAwsBootstrapConfiguration`. STS seems merely to consist of the `security_token` transformation, TTL, Role ARN, and a slightly different path in Vault.

Reply:
Done.